src/basic/copy.c

```diff
@@ -114,9 +112,9 @@ int copy_bytes(int fdf, int fdt, uint64_t max_bytes, bool try_reflink) {
 
                 /* As a fallback just copy bits by hand */
                 {
-                        uint8_t buf[m];
+                        uint8_t buf[MIN(m, COPY_BUFFER_SIZE)];
 
-                        n = read(fdf, buf, m);
+                        n = read(fdf, buf, sizeof buf);
```

Comment:
I'm surprised `sizeof` works without parentheses, and even more so about the fact that we have other locations in the source where it is used that way :)